log/devlog/devlog.go

```diff
@@ -0,0 +1,127 @@
+// Package devlog provides a plain-text logger suitable for reading from the
+// console.
+//
+// For production, the team prefers JSON logging, but plain-text can be useful
+// for local development and tests.
+//
+// So while this package shouldn't be used for production, it can be enabled
+// for your development use by setting TIDEPOOL_USE_DEVLOG=1 as an environment
+// variable. It will also be enabled if POD_NAMESPACE=default (this is set via
+// the development repo when using it's kind k8s cluster).
+
+package devlog
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/tidepool-org/platform/errors"
+	plog "github.com/tidepool-org/platform/log"
+)
+
+// New provides a plain-text log.Logger suitable for development or testing.
+func New(writer io.Writer, levelRanks plog.LevelRanks, level plog.Level) (plog.Logger, error) {
+	if writer == nil {
+		return nil, errors.New("writer is missing")
+	}
+	logLogger := log.New(writer, "", 0)
+	return plog.NewLogger(&serializer{logLogger}, levelRanks, level)
+}
+
+func NewWithDefaults(writer io.Writer) (plog.Logger, error) {
+	return New(writer, plog.DefaultLevelRanks(), plog.DebugLevel)
+}
+
+type serializer struct {
+	*log.Logger
+}
+
+func (s *serializer) Serialize(fields plog.Fields) error {
+	var pairs = make([]string, 0, len(fields))
+	var msg, msgTime, msgLevel, msgCaller string
+	var showCaller = false
+
+	msgLevel = abbreviateLevel("")
+	msgTime = time.Now().Format(time.Stamp)
+
+	for key, value := range fields {
+
+		switch key {
+		case "caller":
+			msgCaller = formatCaller(value)
+			continue
+		case "level":
+			level := sOrPlusV(value)
+			msgLevel = abbreviateLevel(level)
+			if level == "error" {
+				showCaller = true
+			}
+			continue
+		case "message":
+			msg = sOrPlusV(value)
+			continue
+		case "process":
+			// process isn't useful outside of sumo
+			continue
+		case "time":
+			// time is terser and constant width for console logging
+			tv, err := time.Parse(time.RFC3339Nano, sOrPlusV(value))
+			if err != nil {
+				msgTime = sOrPlusV(value)
+				continue
+			}
+			msgTime = tv.Format(time.Stamp)
+			continue
```

Comment:
Rather than all of the `continue` statements in each `case`, why not just put the `pairs = append(pairs, key+"="+sOrPlusV(value))` in a `default` (and then remove the unnecessary `continue`).